crates/cheatcodes/src/inspector.rs

```diff
@@ -1333,25 +1333,30 @@ impl Cheatcodes {
     fn meter_gas(&mut self, interpreter: &mut Interpreter) {
         match &self.gas_metering {
             None => {}
-            // need to store gas metering
+            // Need to store gas metering.
             Some(None) => self.gas_metering = Some(Some(interpreter.gas)),
             Some(Some(gas)) => {
                 match interpreter.current_opcode() {
                     opcode::CREATE | opcode::CREATE2 => {
-                        // set we're about to enter CREATE frame to meter its gas on first opcode
-                        // inside it
+                        // Set we're about to enter CREATE frame to meter its gas on first opcode
+                        // inside it.
                         self.gas_metering_create = Some(None)
                     }
                     opcode::STOP | opcode::RETURN | opcode::SELFDESTRUCT | opcode::REVERT => {
-                        // If we are ending current execution frame, we want to just fully reset gas
-                        // otherwise weird things with returning gas from a call happen
-                        // ref: https://github.com/bluealloy/revm/blob/2cb991091d32330cfe085320891737186947ce5a/crates/revm/src/evm_impl.rs#L190
-                        //
-                        // It would be nice if we had access to the interpreter in `call_end`, as we
-                        // could just do this there instead.
                         match &self.gas_metering_create {
                             None | Some(None) => {
-                                interpreter.gas = Gas::new(0);
+                                // If we are ending current execution frame, we want to reset
+                                // interpreter gas to the value of gas spent during frame, so only
+                                // the consumed gas is erased.
+                                // ref: https://github.com/bluealloy/revm/blob/2cb991091d32330cfe085320891737186947ce5a/crates/revm/src/evm_impl.rs#L190
+                                //
+                                // It would be nice if we had access to the interpreter in
+                                // `call_end`, as we could just do this there instead.
+                                interpreter.gas =
+                                    Gas::new(interpreter.gas.limit() - interpreter.gas.remaining());
```

Comment:
'gas.spent()' ?

Reply:
thanks, missed that, updated in https://github.com/foundry-rs/foundry/pull/8481/commits/2d3788d928b13e2c1c2201b1fca53aee7d3e9ccf